src/dgipy/dgidb.py

```diff
@@ -122,69 +150,123 @@ def get_interactions(
+                pmid: $pmid
+                interactionType: $interactionType
+              ) {
+                nodes {
+                  name
+                  longName
+                  geneCategories {
+                    name
+                  }
+                  interactions {
+                    interactionAttributes {
+                      name
+                      value
+                    }
+                    drug {
+                      name
+                      approved
+                    }
+                    interactionScore
+                    interactionClaims {
+                      publications {
+                        citation
+                        pmid
+                      }
+                      source {
+                        sourceDbName
+                      }
+                    }
+                  }
+                }
+              }
+            }
+            """
         )
     elif search == "drugs":
-        query = (
-            '{\ndrugs(names: ["'
-            + terms.upper()
-            + '"]'
-            + filters
-            + ") {\nnodes{\nname\napproved\ninteractions {\ngene {\nname\n}\ninteractionAttributes {\nname\nvalue\n}\ninteractionScore\ninteractionClaims {\npublications {\npmid\ncitation\n}\nsource {\nfullName\nid\n}\n}\n}\n}\n}\n}"
+        query = gql(
+            """
+            query getInteractionsByDrug($names: [String!], $immunotherapy: Boolean, $antineoplastic: Boolean, $sourceDbName: String, $pmid: Int, $interactionType: String, $approved: Boolean) {
+              drugs(
+                names: $names
+                immunotherapy: $immunotherapy
+                antiNeoplastic: $antineoplastic
+                sourceDbName: $sourceDbName
+                pmid: $pmid
+                interactionType: $interactionType
+                approved: $approved
+              ) {
+                nodes {
+                  name
+                  approved
+                  interactions {
+                    interactionAttributes {
+                      name
+                      value
+                    }
+                    gene {
+                      name
+                    }
+                    interactionScore
+                    interactionClaims {
+                      publications {
+                        citation
+                        pmid
+                      }
+                      source {
+                        sourceDbName
+                      }
+                    }
+                  }
+                }
+              }
+            }
+            """
         )
     else:
         msg = "Search type must be specified using: search='drugs' or search='genes'"
         raise Exception(msg)
 
     api_url = api_url if api_url else API_ENDPOINT_URL
-    r = requests.post(api_url, json={"query": query}, timeout=20)
+    client = _get_client(api_url)
+    result = client.execute(query, variable_values=params)
```

Comment:
big fan of this, def feels better than building a giant query string and sending it with requests